android/engine/src/main/java/org/smartregister/fhircore/engine/ui/login/LoginScreen.kt

```diff
@@ -251,20 +251,31 @@ fun LoginPage(
           }
         )
         Spacer(modifier = modifier.height(10.dp))
-        if (loginError.isNotEmpty()) {
-          Text(
-            fontSize = 14.sp,
-            color = MaterialTheme.colors.error,
-            text = stringResource(id = R.string.login_error, loginError),
-            modifier =
-              modifier
-                .wrapContentWidth()
-                .padding(0.dp)
-                .align(Alignment.Start)
-                .testTag(LOGIN_ERROR_TEXT_TAG)
-          )
-        }
-        Spacer(modifier = modifier.height(40.dp))
+        Text(
+          fontSize = 14.sp,
+          color = MaterialTheme.colors.error,
+          text =
+            when (loginErrorState) {
+              LoginErrorState.NO_ERROR -> ""
```

Comment:
do we need the no_error state? could we handle this with an `else` for the `when`?

Reply:
then whenever we set to this constant we'd set to null instead

Reply:
We do not need it actually. I will make the `loginErrorState` parameter nullable.

Reply:
I have explicitly handled `null` condition in the `when` statement; we may in future need to further categorize the error messages.

Reply:
ok good call, we wouldn't want to bury an error if it doesn't explicitly match the when conditions